src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java

```diff
@@ -285,7 +286,12 @@ public Parent getNode() {
 
     @FXML
     private void addNewFile() {
-        viewModel.addNewFile();
+        LinkedFileDialogController controller = new LinkedFileDialogController();
+
+        controller.showAndWait().ifPresent(result -> {
```

Comment:
You have to use `org.jabref.gui.DialogService#showCustomDialogAndWait(javafx.scene.control.Dialog<R>)` to get the right styling for the dialog.